loadbalancer/algorithm.go

```diff
@@ -203,12 +105,16 @@ func (r *random) Apply(ctx *routing.LBContext) routing.LBEndpoint {
 		return ctx.Route.LBEndpoints[0]
 	}
 
-	i := r.rand.Intn(len(ctx.Route.LBEndpoints))
-	if ctx.Route.LBFadeInDuration <= 0 {
-		return ctx.Route.LBEndpoints[i]
+	ctx.Route.Lock()
+	defer ctx.Route.Unlock()
+
+	endpoints := ctx.Route.GoodEndpoints
+	if len(endpoints) == 0 {
+		endpoints = ctx.Route.LBEndpoints
 	}
+	i := r.rand.Intn(len(endpoints))
```

Comment:
Here we miss locking as rand.Intn is not thread safe.

Reply:
https://github.com/zalando/skipper/pull/2412
I make `rand.Intn` thread safe in this PR.